tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -832,6 +837,11 @@ struct DxcPdbUtils : public IDxcPdbUtils, public IDxcPixDxilDebugInfoFactory
     *ppVersionInfo = result.Detach();
     return S_OK;
   }
+
+  virtual HRESULT STDMETHODCALLTYPE SetCompiler(_In_ IDxcCompiler3 *pCompiler) {
+    m_pCompiler = pCompiler;
```

Comment:
Should we fail if IsFullPDB() is true (and pCompiler is non-null)?  That would catch someone setting the compiler too late for it to be used.